### PR TITLE
refactor updateDeliveryDate controller

### PR DESCRIPTION
**Pull request by SashkoStahovskyi**

### CODE REVIEW

File: goods-partner-backend/src/main/java/com/goodspartner/action/OrderAction.java [hunk 1]
```diff
@@ -25,6 +18,11 @@ public void perform(OrderExternal order) {
             order.setDropped(false);
             order.setExcluded(false);
         }
+
+        @Override
+        public void performForList(List<OrderExternal> orderExternals) {
+            orderExternals.forEach(this::perform);
+        }
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/action/OrderAction.java:25]: I think we could avoid this and have this code in the invocation method

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java [hunk 2]
```diff
@@ -159,17 +160,20 @@ public List<OrderDto> getFilteredOrders(boolean excluded, boolean dropped) {
 
     @Transactional
     @Override
-    public OrderDto updateDeliveryDate(int orderId, LocalDate deliveryDate, OrderAction action) {
-        OrderExternal orderExternal = orderExternalRepository.findById(orderId)
-                .orElseThrow(() -> new OrderNotFoundException(orderId));
+    public List<OrderDto> updateDeliveryDate(UpdateDto updateDto, OrderAction orderAction) {
+        List<Integer> ordersIdList = updateDto.getOrdersIdList();
+        List<OrderExternal> ordersExternalsList = orderExternalRepository.findAllById(ordersIdList);
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java:165]: findAllById
- SELECT .. FROM .. WHERE ID IN (...)

Comment (belosh59) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java:165]: jpa:
    properties:
      hibernate:
        format_sql: false
        show_sql: false

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java [hunk 3]
```diff
@@ -159,17 +160,20 @@ public List<OrderDto> getFilteredOrders(boolean excluded, boolean dropped) {
...
+        ordersExternalsList.stream().findFirst().orElseThrow(OrderNotFoundException::new);
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java:166]: lets just remove

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java [hunk 4]
```diff
@@ -159,17 +160,20 @@ public List<OrderDto> getFilteredOrders(boolean excluded, boolean dropped) {
...
 
-        orderExternal.setDeliveryDate(deliveryDate);
-        action.perform(orderExternal);
+        LocalDate date = updateDto.getDeliveryDate();
+        ordersExternalsList.forEach(order -> order.setDeliveryDate(date));
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java:169]: ordersExternalsList.forEach(order -> {
   action.perform(order);
   order.setDeliveryDate(updateDto.getDeliveryDate())
});

File: goods-partner-backend/src/main/java/com/goodspartner/action/OrderAction.java (hunk 1 above)
Comment (SashkoStahovskyi) [reply goods-partner-backend/src/main/java/com/goodspartner/action/OrderAction.java:25]: fix

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java (hunk 4 above)
Comment (SashkoStahovskyi) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java:169]: fix

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java [hunk 5]
```diff
@@ -159,17 +160,20 @@ public List<OrderDto> getFilteredOrders(boolean excluded, boolean dropped) {
...
 
-        Optional<Delivery> delivery = deliveryRepository.findByStatusAndDeliveryDate(DeliveryStatus.DRAFT, deliveryDate);
-        delivery.ifPresent(orderExternal::setDelivery);
+        orderAction.performForList(ordersExternalsList);
 
-        return orderExternalMapper.mapOrderExternalToOrderDto(orderExternalRepository.save(orderExternal));
+        Optional<Delivery> delivery = deliveryRepository.findByStatusAndDeliveryDate(DeliveryStatus.DRAFT, date);
+        delivery.ifPresent(deliveryValue -> ordersExternalsList.forEach(order -> order.setDelivery(deliveryValue)));
```
[the remark this answers is not in the document]
Comment (SashkoStahovskyi) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultOrderExternalService.java:174]: fix

**Review by belosh59 — APPROVED** [review]

**Pull request merged**